src/main/java/net/openhft/chronicle/wire/WireMarshaller.java

```diff
@@ -1113,15 +1153,27 @@ protected void setValue(Object o, @NotNull ValueIn read, boolean overwrite) thro
         }
 
         @Override
-        public void getAsBytes(Object o, @NotNull Bytes<?> bytes) throws IllegalAccessException {
-            bytes.writeUtf8(String.valueOf(field.get(o)));
+        protected void setDefaultValue(Object ignored, Object o) {
+            if (defaultValue != null && defaultValue instanceof Resettable && !(defaultValue instanceof DynamicEnum)) {
+                Object existingValue = unsafeGetObject(o, offset);
+
+                if (existingValue == defaultValue)
+                    return;
+
+                if (existingValue != null && existingValue.getClass() == defaultValue.getClass()) {
+                    ((Marshallable) existingValue).reset();
```

Comment:
Could this end in an infinite loop with a self-referential DTO or e.g. a doubly-linked structure?

Not sure how you'd go about avoiding that in a garbage free way, but perhaps something to consider?

Reply:
Yes, it could; however, due to the safety of JVM, we'll just get a neat `StackOverflowError` out of it.

I don't think we could easily guard against this scenario. DTOs are supposed to be trees and not graphs, anyway, since it is assumed they are serialized eventually.